### DISCUSSION TITLE
How to make Rag understand the full text content

### DISCUSSION THREAD

**Discussion by smileyboy2019** [langchain4j/langchain4j#2093] [Ideas]
I found two projects that can convert documents into PDFs first, and then use images to LLM for full-text interpretation. A dozens of pages of PPT can recognize all the content, and a multimodal model can achieve the same effect as graphRAG
https://github.com/PromtEngineer/localGPT-Vision
https://github.com/OpenBMB/VisRAG

**Comment by smileyboy2019** [comment]
![image](https://github.com/user-attachments/assets/65c83fbe-db30-45cc-a925-9eada72e20c7)
Efficient Processing of Long Text: RAPTOR Retrieval Technology and Its Application in RAG
![image](https://github.com/user-attachments/assets/f0317a0d-fdb9-4aae-a597-268b33ae9a68)
Does langchain4j support this method

**Comment by smileyboy2019** [comment]
How to handle long texts and are there any good solutions? Everyone can discuss the following

**Reply by glaforge** [reply]
There are myriads of way to load and chunk documents (I covered some chunking techniques in my [recent talk](https://glaforge.dev/talks/2024/10/14/advanced-rag-techniques/) for example).
But it's also true that for some rich (and big) documents, you may loose some context, or for example miss the images in the PDFs, etc. In that case, for such documents, maybe using multimodal models like Gemini which understand PDF files natively might help achieve better and more accurate results.
